PC/winreg.c

```diff
@@ -564,50 +564,44 @@ Py2Reg(PyObject *value, DWORD typ, BYTE **retDataBuf, DWORD *retDataSize)
 {
     Py_ssize_t i,j;
     switch (typ) {
-        case REG_DWORD:
-            if (value != Py_None && !PyLong_Check(value))
+        case REG_DWORD: {
+            DWORD d = 0;
+            if (PyLong_Check(value)) {
+                d = PyLong_AsUnsignedLong(value);
+                if (d == (DWORD)(-1) && PyErr_Occurred()) {
+                    return FALSE;
+                }
+            } else if (value != Py_None) {
```

Comment:
Please follow PEP 7, ref. section 3.5 in the dev guide:

```suggestion
            }
            else if (value != Py_None) {
```